### PR TITLE
Fix Mac OS compilation

### PR DESCRIPTION
**Pull request by rlakis**

### CODE REVIEW

File: src/config/Config.h [hunk 1]
```diff
@@ -44,7 +44,7 @@ class wxFileConfig;
 
 //! Do not instantiate objects of this class. Use config() function (see below).
 
-#if defined(__UNIX__) && !defined(__WXMAC_OSX__)
```
Comment (arvanus) [inline src/config/Config.h:47]: isn't __WXMAC_OSX__ directive used anymore?
Thank you

**Comment by rlakis** [comment]
on my MacbookPro (Mac OS 13 and 14) it did not match
It looks outdated may be for Carbon Mac

**Comment by arvanus** [comment]
So could you adjust to keep this older directive too?

**Comment by arvanus** [comment]
Don't know if we have users using this OS

**Comment by rlakis** [comment]
With the release of macOS 10.15 Catalina, the Carbon API was officially
discontinued and removed, leaving Cocoa as the sole primary API for
developing macOS applications. Apple Inc.

Current Mac OS version is 14.0

**Pull request merged**